File: assign01/tcpserver.c
/********************
Lab : Network Lab
Name : Abhinav Kumar
Enroll no.: 510817075
Assignment : 01(Q.01)
*********************/
#include <stdio.h>
#include <stdlib.h>
#include <sys/types.h>
#include <sys/socket.h>
#include <netinet/in.h>
int main()
{
    char message[250] = "This message is send by server to client that connection was successful.\n";
    int server_socket;
    server_socket = socket(AF_INET, SOCK_STREAM, 0); //server endpoint
    struct sockaddr_in server_address;
    server_address.sin_family = AF_INET;
    server_address.sin_port = htons(9002);
    server_address.sin_addr.s_addr = INADDR_ANY;
    bind(server_socket, (struct sockaddr *)&server_address, sizeof(server_address)); //binding socket with address
    listen(server_socket, 5);                                                        //n cnonection will be listened and connected
    int client_socket;
    printf("started server.\n");
    while (1)
    {
        client_socket = accept(server_socket, NULL, NULL); //we will get client's endpoint adress from here
        printf("got request.\n");
        send(client_socket, message, sizeof(message), 0); //message was sent to client
        printf("sent response.\n");
        close(client_socket); //server socket closed
    }
    return 0;
}

File: assign01/tcpserver.py
##########################
# Lab : Network Lab      #
# Name : Abhinav Kumar   #
# Enroll no.: 510817075  #
# Assignment : 01(Q.01)  #
##########################
import socket


def tcpserver(host=socket.gethostname(), port=9002):
    # started a tcp socketendpoint
    server_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
    server_socket.bind((host, port))  # allocate address to server
    server_socket.listen(5)  # listen for request
    print(f"started server at : {host}:{port}")
    while True:
        client_socket, client_address = server_socket.accept()  # accepts requests
        print(f"got request from {client_address[0]}:{client_address[1]}")
        message = 'you will get this message if connection is successful'
        # decoding bytecoded message to readable
        client_socket.send(message.encode('ascii'))
        print(f"sent response to {client_address[0]}:{client_address[1]}")
        client_socket.close()  # closing client connection


if __name__ == "__main__":
    # if i want to start server on certain address and port
    # then i have to pass address and port as argument by default it's set to loopback addr
    tcpserver()

# i have used fstring in this assignment if during execution of this code error happens
# then python3 on your system needs upgrade means it's below 3.6
